### PR TITLE
Add gen command for it:av:scan:result (SYN-6661)

### PR DESCRIPTION
**Pull request by mikemoritz**

### CODE REVIEW

File: synapse/lib/stormlib/gen.py [hunk 1]
```diff
@@ -105,10 +105,32 @@ class LibGen(s_stormtypes.Lib):
                        'desc': 'Type normalization will fail silently instead of raising an exception.'},
                   ),
                   'returns': {'type': 'node', 'desc': 'A lang:language node with the given code.'}}},
+        {'name': 'itAvScanResultByTarget',
+         'desc': 'Returns an it:av:scan:result node by deconflicting with a target and signature name, adding the node if it does not exist.',
+         'type': {'type': 'function', '_funcname': '_storm_query',
+                  'args': (
+                      {'name': 'type', 'type': 'str', 'desc': 'The target type.'},
+                      {'name': 'target', 'type': 'str', 'desc': 'The target value.'},
```
Comment (vEpiphyte) [inline synapse/lib/stormlib/gen.py:113]: Calling convention and docs need to be reconciled. Order not the same and there appear to be three required arguments?

```
function itAvScanResultByTarget(signame, type, targ, scanner=$lib.null, time=$lib.null, try=$lib.false)
```

vs

```
cli> storm $lib.print($lib.gen.itAvScanResultByTarget(type='someType', target='someTarget'))
Executing query at 2024/01/26 14:06:17.634
....
ERROR: ('StormRuntimeError', {'efile': 'ast.py', 'eline': 4758, 'mesg': 'itAvScanResultByTarget() missing required argument signame', 'highlight': {'hash': '09a9d89d4cc5ac22a976aeba5f80c07f', 'lines': (212, 212), 'columns': (40, 112), 'offsets': (6104, 6176)}})
```

File: synapse/lib/stormlib/gen.py [hunk 2]
```diff
@@ -105,10 +105,32 @@ class LibGen(s_stormtypes.Lib):
...
+                       'desc': 'An optional time when the scan was run to include in the deconfliction.'},
+                      {'name': 'try', 'type': 'boolean', 'default': False,
+                       'desc': 'Type normalization will fail silently instead of raising an exception.'},
+                  ),
+                  'returns': {'type': 'node', 'desc': 'An it:av:scan:result node.'}}},
     )
     _storm_lib_path = ('gen',)
 
     _storm_query = '''
+        function _maybeCast(try, type, valu) {
```
Comment (vEpiphyte) [inline synapse/lib/stormlib/gen.py:127]: Can we mark this as private so we're not exposing it unwittingly. ``__maybeCast``  would do.

File: synapse/lib/stormlib/gen.py (hunk 1 above)
Comment (mikemoritz) [reply synapse/lib/stormlib/gen.py:113]: Fixed

**Review by vEpiphyte — APPROVED** [review]

File: synapse/lib/stormlib/gen.py [hunk 3]
```diff
@@ -507,4 +548,38 @@ class LibGen(s_stormtypes.Lib):
...
+
+                // Also deconflict by scanner name and scan time
+                gen.it.av.scan.result fqdn vertex.link foosig --scanner-name barscanner --time 2022-11-03
+
+                // bar
+                it:av:filehit#foo | gen.it.av.scan.result file :file :sig:name
+        ''',
+        'cmdargs': (
+            ('type', {'help': 'The target type.',
+                      'choices': ['file', 'fqdn', 'host', 'ipv4', 'ipv6', 'proc', 'url']}),
```
[the remark this answers is not in the document]
Comment (mikemoritz) [reply synapse/lib/stormlib/gen.py]: Updated to form/value

File: synapse/lib/stormlib/gen.py [hunk 4]
```diff
@@ -507,4 +549,37 @@ class LibGen(s_stormtypes.Lib):
...
+                // Also deconflict by scanner name and scan time
+                gen.it.av.scan.result inet:fqdn fqdn vertex.link foosig --scanner-name barscanner --time 2022-11-03
+
+                // Generate an it:av:scan:result node from an it:av:filehit node
+                it:av:filehit#foo | gen.it.av.scan.result file:bytes :file :sig:name
+        ''',
+        'cmdargs': (
+            ('form', {'help': 'The target form.'}),
+            ('value', {'help': 'The target value.'}),
+            ('signame', {'help': 'The signature name'}),
```
Comment (Cisphyx) [inline synapse/lib/stormlib/gen.py:574]:
```suggestion
            ('signame', {'help': 'The signature name.'}),
```

**Review by Cisphyx — APPROVED** [review]

File: synapse/lib/stormlib/gen.py [hunk 5]
```diff
@@ -105,10 +105,32 @@ class LibGen(s_stormtypes.Lib):
...
+        {'name': 'itAvScanResultByTarget',
+         'desc': 'Returns an it:av:scan:result node by deconflicting with a target and signature name, adding the node if it does not exist.',
+         'type': {'type': 'function', '_funcname': '_storm_query',
+                  'args': (
+                      {'name': 'form', 'type': 'str', 'desc': 'The target form.'},
+                      {'name': 'value', 'type': 'str', 'desc': 'The target value.'},
+                      {'name': 'signame', 'type': 'str', 'desc': 'The signature name.'},
+                      {'name': 'scanner', 'type': 'str', 'default': None,
+                       'desc': 'An optional scanner software name to include in deconfliction.'},
+                      {'name': 'time', 'type': 'time',
```
Comment (invisig0th) [inline synapse/lib/stormlib/gen.py:117]: being optional, does this need a default?

Comment (vEpiphyte) [reply synapse/lib/stormlib/gen.py]: fixed

**Pull request merged**